### PR TITLE
Depend on rlang

### PR DESCRIPTION
**Pull request by lionel-**

### CODE REVIEW

File: R/revdepcheck.R [hunk 1]
```diff
@@ -134,7 +134,7 @@ revdep_install <- function(pkg = ".", quiet = FALSE) {
     with_libpaths(
       dir_find(pkg, "old"),
       with_options(
-        list(warn = 2),
+        warn = 2,
```
Comment (gaborcsardi) [inline R/revdepcheck.R]: This is not good, I think, see `withr::with_options`.

File: R/revdepcheck.R [hunk 2]
```diff
@@ -148,7 +148,7 @@ revdep_install <- function(pkg = ".", quiet = FALSE) {
     with_libpaths(
       dir_find(pkg, "new"),
       with_options(
-        list(warn = 2),
+        warn = 2,
```
Comment (gaborcsardi) [inline R/revdepcheck.R]: Same here.

File: R/utils.R [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-`%||%` <- function(l, r) if (length(l) == 0) r else l
-
-is_string <- function(x) {
-  is.character(x) && length(x) == 1 && !is.na(x)
+`%|0|%` <- function(x, y) {
+  if (!length(x)) y else x
 }
 
-#' @importFrom utils installed.packages
-
```
Comment (gaborcsardi) [inline R/utils.R]: Why remove this?

File: R/utils.R [hunk 4]
```diff
@@ -1,23 +1,21 @@
+#' @import rlang
+#' @importFrom utils installed.packages
 
-`%||%` <- function(l, r) if (length(l) == 0) r else l
-
-is_string <- function(x) {
-  is.character(x) && length(x) == 1 && !is.na(x)
```
Comment (gaborcsardi) [inline R/utils.R]: Why remove `is_string`?

File: R/utils.R [hunk 5]
```diff
@@ -67,14 +65,14 @@ str_trunc <- function(x, n) {
...
-#' @importFrom withr with_options with_libpaths with_envvar
+#' @importFrom withr with_libpaths with_envvar
 
 execute_r <- function(px_opts, new_session = FALSE) {
   if (new_session) {
     do.call(r, px_opts)
   } else {
     with_options(
-      list(repos = px_opts$repos),
+      repos = px_opts$repos,
```
Comment (gaborcsardi) [inline R/utils.R]: Same here.

File: R/utils.R (hunk 4 above)
Comment (lionel-) [reply R/utils.R]: Using rlang one.

File: R/revdepcheck.R (hunk 1 above)
Comment (lionel-) [reply R/revdepcheck.R]: The rlang version take spliceable dots.

File: R/utils.R (hunk 3 above)
Comment (lionel-) [reply R/utils.R]: I'm not sure, maybe a typeo, I'll double check.

File: R/revdepcheck.R (hunk 1 above)
Comment (gaborcsardi) [reply R/revdepcheck.R]: Oh, so rlang has  `with_` functions, I did not know that.  I am pretty sure that I miss this at some point in the future, can please add an explicit `rlang::` here, and below?

Comment (lionel-) [reply R/revdepcheck.R]: Done.

**Comment by gaborcsardi** [comment]
OK, thanks!

**Pull request merged**